cql-parser/src/main/scala/troy/cql/parser/CqlParser.scala

```diff
@@ -53,7 +55,13 @@ object CqlParser extends JavaTokenParsers
    */
   def constant: Parser[Constant] = {
     import Constants._
-    (string | uuid | number | boolean) ^^ Constant // | hex // TODO
+    def str = (string | uuid) ^^ StringConstant
+    def int = integer ^^ { s => new IntegerConstant(s.toInt) }
+    def floatNum = float ^^ { s => new FloatConstant(s.toFloat) }
+    //def uuid = uuid ^^ { s => new UuidConstant(new UUID (UUID.fromString(s))) }
+    def bool = boolean ^^ { s => new BooleanConstant(s.toBoolean) }
+
+    str | int | floatNum | bool
```

Comment:
Try swapping `int` and `floatNum`
This should fix the failing test case related to FloatConstant. This is due to the fact that order matters, as int matches on the float numbers till it reaches the dot, so it gives the error without backtracking